### PR TITLE
COMM-1182: Increase delay accuracy

### PR DESCRIPTION
**Pull request by codyherzog-zz**
Make timer frequency much faster to improve delay simulation accuracy.

I tested delay, loss, and cap disturbances, and they all seem to work as well or better than before these changes.

I think CPU usage is increased slightly.

### CODE REVIEW

**Comment by codyherzog-zz** [comment]
Hello, @christopher-coffin 

I think it would be good for someone on the COMM team to review this PR before I merge.

Please delegate as desired.

File: src/divert.c
Comment (agnickolov) [inline src/divert.c:11]: I believe this value is too low. For example this will trigger excessive logging due to the condition on line 232 being always true. It'll also lead to elevated CPU utilization since the timer thread will essentially be busy looping most of the time. The only reason this won't cause real issues is because the Windows scheduler does not obey it and can easily sleep for 10ms or more at a time. As a practical suggestion, I'd recommend a value of 10ms instead.

**Review by agnickolov — APPROVED** [review]
On the whole I disapprove of the polling mechanism being used. Windows has a far more efficient mechanism to implement high performance I/O operations - I/O Completion Ports. However, this being open source code which we are unlikely to rewrite, the present changes seem to be ok, with the caveat that the sleep time chosen of 1ms is too low as I already commented.

File: src/divert.c
Comment (codyherzog-zz) [reply src/divert.c:11]: Hello.

I ran some ping tests to compare the accuracy of simulated delays between a CLOCK_WAITMS value of 10 ms and 1 ms.

On my test system, with the 10 ms value, the simulated delays are accurate within about 6 ms.

With the 1 ms value, the accuracy increases to about 2 ms.

I initially shared your concerns about increased CPU usage, but my testing so far has shown that it seems OK.

Using the 1 ms CLOCK_WAITMS value, I compared the CPU load in a session with and without Clumsy by monitoring the task manager CPU performance graph. I could not see any noticeable CPU increase or decrease as I started and stopped Clumsy.

Of course, things may behave differently on different systems, so the steps to test for this Jira ticket will include SQA comparing old and new Clumsy on various different computers, to be sure the accuracy is increased, and the CPU is not increased very much.

Does that sound good?

**Comment by codyherzog-zz** [comment]
> On the whole I disapprove of the polling mechanism being used.

I agree that the polling seems inefficient, and I also agree that it doesn't make sense to rewrite/refactor at this time.

The main purpose of this pull request is to increase the delay accuracy without causing a noticeable increase in CPU.

File: src/divert.c
Comment (codyherzog-zz) [reply src/divert.c:11]: @agnickolov You've already approved the PR, but I'll hold off on merging until you have a chance to respond to my comment above. Thanks.

Comment (agnickolov) [reply src/divert.c:11]: Sure, go ahead.

**Pull request merged**
